stacks-signer/src/ping/periodic_pinger.rs

```diff
@@ -0,0 +1,102 @@
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        mpsc::Sender,
+        Arc,
+    },
+    thread::{self, JoinHandle},
+    time::Duration,
+};
+
+use crate::runloop::RunLoopCommand;
+use tracing::{info, trace, Span};
+
+/// Stop the pinger thread
+pub struct ThreadStopHandle {
```

Comment:
Nit: can you name this something specific to the ping thread?  Like `PingStopHandle`?